File: Interview-Prep/string/Character-filtering/242-optimized/242.cpp
#include<iostream>
#include<vector>
#include<unordered_map>
using namespace std;
class Solution{
    public:
    bool isAnagram(string s, string t){
        if(s.size()!=t.size()) return false;
        vector<int> frequency(26,0);
        for(int i=0; i<s.size(); i++){
            frequency[s[i]-'a']++;
            frequency[t[i]-'a']--;
        }
        for(int i=0; i<26; i++){
            if(frequency[i]!=0) return false;
        }
        return true;
    }
};
int main(){
    Solution solution;
    string s= "anagram", t = "nagaram";
    cout<<solution.isAnagram(s,t);
}

File: Interview-Prep/string/Character-filtering/242-optimized/242.py
class Solution(object):
    def isAnagram(self,s, t):
        if len(t)!=len(s):
            return False
        frequency=[0]*26
        for i in range(len(s)):
            frequency[ord(s[i])-97]+=1
            frequency[ord(t[i])-97]-=1
        for i in range(0,26):
            if frequency[i]!=0:
                return False
        return True
def main():
    sol =Solution()
    print(sol.isAnagram("ab", "a"))
if __name__=="__main__":
    main()
        